calblueprint/dccentralkitchen-clerks
# Update environment configs to auto-switch on release channel

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

- Created `environment.example` to be duplicated into `environment.js` and added the usage & code that needs to be replaced in the generated `lib/airtable.js` to pull the variables from a new config and deleted the old `.env.example`
    - The new `environment.example` switches on Expo's `releaseChannel`. More details later
- Created `.env.generator.example` to be duplicated into `.env.generator` and updated the generator config to pull from `.env.generator`
- Removed the package from `package.json`
- Updated `.gitignore`

## Related cleanup
- Added `RecordIds` to `constants/` to swap automatically between `dev` and `prod` bases for hardcoded IDs and pulled IDs from there instead of hardcoding
- This only applies for the bypasses anyway, but now it will only show testing bypasses when developing locally
- Navigation for bypasses was broken due to the `react-navigation` upgrade

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

Modified from https://alxmrtnz.com/thoughts/2019/03/12/environment-variables-and-workflow-in-expo.html#comment-4589309119 

### Related PRs

Analogous to https://github.com/calblueprint/dccentralkitchen/pull/95

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

#6 (prev env update)

## How to review

To test the environment updates:
Since we're switching on `manifest.releaseChannel`, the prod variables will only switch automatically if `expo build` or `expo publish` is run with the flag set to prod; i.e `--release-channel prod`

With `--release-channel prod`, nothing should break.

With `--release-channel staging`, nothing should work (LOL). ** We might want to consider defaulting to dev base ID & api keys, but this is a change that only needs to be made in `environment.js` so I think it's more or less ok.

With `production mode` flipped with the Expo `localhost:19002` FE, nothing should work.

You can check that this works by running `expo start` locally for dev mode, and by using the versions I published here (for Android):
https://expo.io/@anniero/dccentralkitchen-clerks-test
https://expo.io/@anniero/dccentralkitchen-clerks-test?release-channel=prod

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

- Team must update to use new environment configs

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

To **reproduce** this locally, you will need an Expo account. If you do not already have one, Expo will prompt you from the terminal after you attempt to publish.

Steps to publish: *copy-pasted from `dccentralkitchen` repo

Update the app config in `app.json`:
![image](https://user-images.githubusercontent.com/21094576/79501971-ce14e100-7fe3-11ea-96bf-847e2a380486.png)

With the Expo frontend, you can click here to change the `app.json` config
![image](https://user-images.githubusercontent.com/21094576/79500862-01567080-7fe2-11ea-85cb-5fd138b86d04.png)

![image](https://user-images.githubusercontent.com/21094576/79502025-e2f17480-7fe3-11ea-90ae-b722cda4e94b.png)

** I couldn't find a way to set `releaseChannel` this way, so I recommend changing it manually.

Run `expo publish --release-channel prod`
** The release channel name _must_ be `prod`, not `production` because that's what `environment.js` expects

You should get to this page (with your own username):
![image](https://user-images.githubusercontent.com/21094576/79502198-24821f80-7fe4-11ea-995d-9b9e52fcfae1.png)

I recommend testing with your phone (it's easier) or click the link to use the Expo simulator.

** Note: 
A post-publish hook will run for Sentry. **It's fine if it fails as is, since it doesn't affect whether the app gets published** If you want to see Sentry logs, you'll need to configure `app.json` according to https://github.com/calblueprint/dccentralkitchen/pull/85 (or look at the Sentry docs). The Sentry configs are also in `environment.example / js`; you will need to get login info to view the logs (in Slack), and the auth token from someone (try leadership) in Blueprint.
CC: @wangannie 

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."
